main.go

```diff
@@ -318,280 +256,253 @@ func main() {
+				bundleIDEntitlementsMap := archive.BundleIDEntitlementsMap()
+				bundleIDs := []string{}
+				for bundleID := range bundleIDEntitlementsMap {
+					bundleIDs = append(bundleIDs, bundleID)
+				}
 
 				fmt.Println()
-				log.Printf("Target Bundel ID - Entitlements map")
-				idx := -1
-				for bundleID, entitlements := range bundleIDEntitlemnstMap {
-					idx++
+				log.Printf("Target Bundle ID - Entitlements map")
+				for bundleID, entitlements := range bundleIDEntitlementsMap {
 					entitlementKeys := []string{}
 					for key := range entitlements {
 						entitlementKeys = append(entitlementKeys, key)
 					}
-					log.Printf("%s: [%s]", bundleID, strings.Join(entitlementKeys, " "))
+					log.Printf("%s: %s", bundleID, entitlementKeys)
 				}
 
-				codeSignGroups := export.ResolveCodeSignGroups(installedCertificates, installedProfiles, bundleIDEntitlemnstMap)
-
 				fmt.Println()
-				log.Printf("Installed codesign settings")
-				for idx, group := range codeSignGroups {
-					printCodesignGroup(group)
-					if idx < len(codeSignGroups)-1 {
-						fmt.Println()
-					}
-				}
+				log.Printf("Resolving CodeSignGroups...")
 
+				codeSignGroups := export.CreateSelectableCodeSignGroups(certs, profs, bundleIDs)
 				if len(codeSignGroups) == 0 {
-					log.Errorf("Failed to find code singing groups for specified export method (%s)", exportMethod)
+					log.Errorf("Failed to find code signing groups for specified export method (%s)", exportMethod)
 				}
 
-				// Filter for specified export team
-				if len(codeSignGroups) > 0 && configs.TeamID != "" {
-					log.Warnf("Filtering CodeSignInfo groups for export method: %s...", exportMethod)
-
-					codeSignGroups = export.FilterCodeSignGroupsForExportMethod(codeSignGroups, exportMethod)
-
-					if len(codeSignGroups) == 0 {
-						log.Errorf("Failed to find code singing groups for specified export method (%s)", exportMethod)
-					}
+				for _, group := range codeSignGroups {
+					log.Debugf(group.String())
 				}
 
-				// Handle if archive used NON xcode managed profile
-				if len(codeSignGroups) > 0 && !archiveCodeSignIsXcodeManaged {
-					log.Warnf("App was signed with NON xcode managed profile when archiving,")
-					log.Warnf("only NOT xcode managed profiles are allowed to sign when exporting the archive.")
-					log.Warnf("Removing xcode managed CodeSignInfo groups")
-
-					codeSignGroups = export.FilterCodeSignGroupsForNotXcodeManagedProfiles(codeSignGroups)
+				filters := []export.SelectableCodeSignGroupFilter{}
 
-					if len(codeSignGroups) == 0 {
-						log.Errorf("Failed to find code singing groups for specified export method (%s) and WITHOUT xcode managed profiles", exportMethod)
-					}
-				}
+				if len(bundleIDEntitlementsMap) > 0 {
+					log.Warnf("Filtering CodeSignInfo groups for target capabilities")
+					filters = append(filters,
+						export.CreateEntitlementsSelectableCodeSignGroupFilter(bundleIDEntitlementsMap))
```

Comment:
close this if here, since export method filer is not related to the entitlements.
also the filter handles correctly len(bundleIDEntitlementsMap) > 0, i would recommend to remove the if.